scripts/kind-build-test.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+set -eo pipefail
+OPTIND=1
+AEMM_URL="amazon-ec2-metadata-mock-service.default.svc.cluster.local"
+AEMM_VERSION="1.2.0"
+AEMM_DL_URL="https://github.com/aws/amazon-ec2-metadata-mock/releases/download/v${AEMM_VERSION}/amazon-ec2-metadata-mock-${AEMM_VERSION}.tgz"
+CLUSTER_NAME_BASE="test"
+DELETE_CLUSTER_ARGS=""
+K8S_VERSION="1.16"
+OVERRIDE_PATH=0
+PRESERVE=false
+PROVISION_CLUSTER_ARGS=""
+START=$(date +%s)
+SCRIPT_PATH="$( cd "$(dirname "$0")" ; pwd -P )"
+ROOT_DIR="$SCRIPT_PATH/../"
+TMP_DIR=""
+# VERSION is the source revision that executables and images are built from.
+VERSION=$(git describe --tags --always --dirty || echo "unknown")
+
+function timeout() { perl -e 'alarm shift; exec @ARGV' "$@"; }
+
+function relpath() {
+  perl -e 'use File::Spec; print File::Spec->abs2rel(@ARGV) . "\n"' "${1}" "${2}"
+}
+
+function clean_up {
+    if [[ "$PRESERVE" == false ]]; then
+        "${SCRIPT_PATH}"/delete-kind-cluster.sh $DELETE_CLUSTER_ARGS || :
+        return
+    fi
+    echo "To resume test with the same cluster use: \"-c $TMP_DIR\""""
+}
+
+function exit_and_fail {
+    local pod_id=$(get_nth_worker_pod || :)
+    kubectl logs "${pod_id}" --namespace kube-system || :
+    END=$(date +%s)
+    echo "⏰ Took $(expr "${END}" - "${START}")sec"
+    echo "❌ NTH Integration Test FAILED $CLUSTER_NAME! ❌"
+    exit 1
+}
+
+function get_nth_worker_pod {
+    kubectl get pods -n kube-system \
+      --selector 'app.kubernetes.io/name=aws-node-termination-handler' \
+      --field-selector="spec.nodeName=$CLUSTER_NAME-worker,status.phase=Running" \
+      --sort-by=.metadata.creationTimestamp \
+      --output jsonpath='{.items[-1].metadata.name}'
+}
```

Comment:
The above bash function is node-termination-handler specific and not related to ACK.